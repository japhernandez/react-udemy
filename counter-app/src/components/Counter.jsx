import React, { useState } from "react";
import PropTypes from "prop-types";

const Counter = ({ value = 10 }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => setCounter(counter + 1);

  const handleReset = () => setCounter(value);

  const handleSubstrac = () => setCounter(counter - 1);

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>

      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleSubstrac}>-1</button>
    </>
  );
};

Counter.propTypes = {
  value: PropTypes.number,
};

export default Counter;
