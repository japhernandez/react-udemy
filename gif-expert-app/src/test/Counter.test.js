import React from "react";
import { shallow } from "enzyme";
import Counter from "../components/Counter";

describe("Testing CounterApp", () => {
  let wrapper, counterText;

  beforeEach(() => {
    wrapper = shallow(<Counter value={10} />);
    counterText = wrapper.find("h2").text().trim();
  });

  test("should render Counter component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should show by default the value of 10", () => {
    expect(counterText).toBe("10");
  });

  test("should must increase with the +1 button to the initial value", () => {
    // find : Search by button element
    // at: Bring the first button
    // simulate: Simulation button of click
    wrapper.find("button").at(0).simulate("click");
    const counterText = wrapper.find("h2").text().trim();
    expect(counterText).toBe("11");
  });

  test("should you must subtract the initial value with the -1 button", () => {
    // find : Search by button element
    // at: Bring the first button
    // simulate: Simulation button of click
    wrapper.find("button").at(2).simulate("click");
    const counterText = wrapper.find("h2").text().trim();
    expect(counterText).toBe("9");
  });

  test("should it must put the default value when clicking on the reset button", () => {
    // find : Search by button element
    // at: Bring the first button
    // simulate: Simulation button of click
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(1).simulate("click");
    const counterText = wrapper.find("h2").text().trim();
    expect(counterText).toBe("10");
  });
});
