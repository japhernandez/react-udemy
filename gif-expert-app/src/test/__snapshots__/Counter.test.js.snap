// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CounterApp should I must render the Counter component 1`] = `ShallowWrapper {}`;

exports[`Testing CounterApp should render Counter component 1`] = `ShallowWrapper {}`;
